cuvec/CMakeLists.txt

```diff
@@ -26,6 +26,39 @@ else()
     endif()
   endif()
 endif()
+if(CUDAToolkit_FOUND)
+  # work-around for https://gitlab.kitware.com/cmake/cmake/-/issues/24921 to
+  # detect local GPU architectures
+  # (inspired by ${CMAKE_ROOT}/Modules/CMakeTestCUDACompiler.cmake)
+  message(WARNING "overriding CMAKE_CUDA_ARCHITECTURES")
```

Comment:
not guaranteed to always exist:

- `${CMAKE_ROOT}/Modules/CMakeTestCUDACompiler.cmake`
- `${CMAKE_ROOT}/Modules/CMakeDetermineCompilerABI.cmake`
- `${CMAKE_CUDA_ARCHITECTURES_ALL}`

also overriding potentially user-specified `CMAKE_CUDA_ARCHITECTURES` isn't ideal